specification/purviewdatagovernance/PurviewDataCatalog/models.tsp

```diff
@@ -0,0 +1,1291 @@
+  id?: uuid;
+
+  @doc("The name of the critical data element.")
+  name?: string;
+
+  @doc("The system data of the critical data element.")
+  systemData?: CatalogModelSystemDataWithExpired;
+
+  @doc("The unique identifier of the critical data element.")
+  domain?: uuid;
+
+  @doc("The description of the critical data element.")
+  description?: string;
+
+  @doc("The contacts associated with the critical data element.")
+  contacts?: Record<CatalogModelContactsValueInner[]>;
+
+  @doc("The tags associated with the critical data element.")
+  managedAttributes?: CatalogModelManagedAttribute[];
+}
+
+@doc("EntityCategory")
+union EntityCategory {
+  @doc("The DOMAIN associated with the critical data element.")
+  Domain: "DOMAIN",
+
+  @doc("The DATAPRODUCT associated with the critical data element.")
+  DataProduct: "DATAPRODUCT",
+
+  @doc("The TERM associated with the critical data element.")
+  Term: "TERM",
+
+  @doc("The DATAASSET associated with the critical data element.")
+  DataAsset: "DATAASSET",
+
+  @doc("The OBJECTIVE associated with the critical data element.")
+  Objective: "OBJECTIVE",
+
+  @doc("The KEYRESULT associated with the critical data element.")
+  KeyResult: "KEYRESULT",
+
+  @doc("The CRITICALDATAELEMENT associated with the critical data element.")
+  CriticalDataElement: "CRITICALDATAELEMENT",
+
+  @doc("The CRITICALDATACOLUMN associated with the critical data element.")
+  CriticalDataColumn: "CRITICALDATACOLUMN",
+
+  @doc("The CUSTOMMETADATA associated with the critical data element.")
+  CustomMetadata: "CUSTOMMETADATA",
+
+  @doc("The ATTRIBUTE associated with the critical data element.")
+  Attribute: "ATTRIBUTE",
+
+  @doc("The ATTRIBUTEINSTANCE associated with the critical data element.")
+  AttributeInstance: "ATTRIBUTEINSTANCE",
+
+  @doc("The WORKFLOW associated with the critical data element.")
+  Workflow: "WORKFLOW",
+
+  @doc("The CATALOGSNAPSHOT associated with the critical data element.")
+  CatalogSnapshot: "CATALOGSNAPSHOT",
+
+  @doc("The WORKFLOWRUN associated with the critical data element.")
+  WorkflowRun: "WORKFLOWRUN",
+
+  string,
+}
+
+@doc("CatalogApiServiceOrderBy.")
+model CatalogApiServiceOrderBy {
+  @doc("The field to order by.")
+  field?: string;
+
+  @doc("The direction of the order.")
+  direction?: string;
+}
+
+@doc("SharedSearchManageAttributeSearchFilter.")
+model SharedSearchManageAttributeSearchFilter {
+  @doc("The field to filter by.")
+  field?: string;
+
+  @doc("The operator to use for the filter.")
+  operator?: string;
+
+  @doc("The value to filter by.")
+  value?: string;
+
+  @doc("The type of the value.")
+  type?: string;
+}
+
+@doc("CriticalDataElementQueryRequest.")
+model CriticalDataElementQueryRequest {
+  @doc("To Skip how many records.")
+  skip?: SkipQueryParameter;
+
+  @doc("To return how many records.")
+  top?: TopQueryParameter;
+
+  @doc("To order the records.")
+  orderby?: CatalogApiServiceOrderBy[];
```

Comment:
Question, is this `orderby` request query, or request body? (similar question to the `top`, `skip` above.

You do not have `@query` over them (when you write them without Core parameter). Hence I am not sure whether it was a mistake, or your existing backend has them in body, not query.

And yes, the recommendation is that these be as query parameter.